### PR TITLE
adding info script for locks

### PR DESCRIPTION
**Pull request by dcritch**
Adding a script to list out all `namespace/pod`'s with locks.

### CODE REVIEW

File: info/locks [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
...LXYgUElEKTsgZG8KICBjb3VudD0kKGVjaG8gJGxpbmUgfCBhd2sgJ3twcmludCAkMX0nKTsKICBwaWQ9JChlY2hvICRsaW5lIHwgYXdrICd7cHJpbnQgJDJ9Jyk7CiAgb3JpZ19waWQ9JHBpZAogIHBwaWQ9JChncmVwIFBQaWQgL3Byb2MvJHtwaWR9L3N0YXR1cyB8IGF3ayAne3ByaW50ICQyfScpCiAgd2hpbGUgW1sgJHBwaWQgLWd0IDEgXV07IGRvCiAgICBwaWQ9JHBwaWQKICAgIHBwaWQ9JChncmVwIFBQaWQgL3Byb2MvJHtwaWR9L3N0YXR1cyB8IGF3ayAne3ByaW50ICQyfScpCiAgZG9uZQogIGlmIFtbICRwcGlkIC1lcSAxIF1dOyB0aGVuCiAgICBwcGlkPSRwaWQKICBmaQogIGlmIFtbICQocHMgLWhwICRwcGlkIC1vIGNtZCB8IGdyZXAgLWMgY29ubW9uKSAtZXEgMSBdXTsgdGhlbgogICAgbnM9JChwcyAtaHAgJHBwaWQgLW8gY21kIHwgZ3JlcCBjb25tb24gfCBhd2sgJ3twcmludCAkOX0nIHwgYXdrIC1GLyAne3ByaW50ICQ1fScgfCBhd2sgLUZfICd7cHJpbnQgJDF9JykKICAgIHBvZD0kKHBzIC1ocCAkcHBpZCAtbyBjbWQgfCBncmVwIGNvbm1vbiB8IGF3ayAne3ByaW50ICQ5fScgfCBhd2sgLUYvICd7cHJpbnQgJDV9JyB8IGF3ayAtRl8gJ3twcmludCAkMn0nKQogICAgaWYgWyAke25zX3BvZHNbIiR7bnN9LyR7cG9kfSJdfSBdOyB0aGVuCiAgICAgIG5zX3BvZHNbIiR7bnN9LyR7cG9kfSJdPWBleHByICR7bnNfcG9kc1siJHtuc30vJHtwb2R9Il19ICsgJGNvdW50YAogICAgZWxzZQogICAgICBuc19wb2RzWyIke25zfS8ke3BvZH0iXT0kY291bnQKICAgIGZpCiAgZmkKZG9uZQpmb3IgcG9kIGluICIkeyFuc19wb2RzW0BdfSI7IGRvCiAgZWNobyAkcG9kICR7bnNfcG9kc1skcG9kXX0KZG9uZSB8IHNvcnQgLW5yIC1rMiB8IGNvbHVtbiAtdCAKCklGUz0kT1JJR19JRlMK
```
Comment (patrickeasters) [inline info/locks:6]: For the sake of readability, could we include the script non-encoded here? Something like this should behave the same, but also be clearer for review/version control/etc:
```bash
SCRIPT=$(cat <<'EOF' | base64 -w 0
#!/usr/bin/env bash

declare -A ns_pods
ORIG_IFS=$IFS
IFS=$(echo -en "\n\b")

for line in $(sudo lslocks | egrep -v '(unknown)' | awk '{print $2}' | sort -nr | uniq -c | sort -nr | egrep -v 'unknown|-1' | grep -v PID); do
  count=$(echo $line | awk '{print $1}');
  pid=$(echo $line | awk '{print $2}');
  orig_pid=$pid
  ppid=$(grep PPid /proc/${pid}/status | awk '{print $2}')
  while [[ $ppid -gt 1 ]]; do
    pid=$ppid
    ppid=$(grep PPid /proc/${pid}/status | awk '{print $2}')
  done
  if [[ $ppid -eq 1 ]]; then
    ppid=$pid
  fi
  if [[ $(ps -hp $ppid -o cmd | grep -c conmon) -eq 1 ]]; then
    ns=$(ps -hp $ppid -o cmd | grep conmon | awk '{print $9}' | awk -F/ '{print $5}' | awk -F_ '{print $1}')
    pod=$(ps -hp $ppid -o cmd | grep conmon | awk '{print $9}' | awk -F/ '{print $5}' | awk -F_ '{print $2}')
    if [ ${ns_pods["${ns}/${pod}"]} ]; then
      ns_pods["${ns}/${pod}"]=`expr ${ns_pods["${ns}/${pod}"]} + $count`
    else
      ns_pods["${ns}/${pod}"]=$count
    fi
  fi
done
for pod in "${!ns_pods[@]}"; do
  echo $pod ${ns_pods[$pod]}
done | sort -nr -k2 | column -t 

IFS=$ORIG_IFS
EOF
)
```

File: info/locks [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+[ -z ${UTILSFILE} ] && source $(echo "$(dirname ${0})/../utils")
+if oc auth can-i debug node >/dev/null 2>&1; then
+  msg "Checking for locks by pod, per node (${BLUE}using oc debug, it can take a while${NOCOLOR})"
+    fw_errors=0
+  # shellcheck disable=SC2016
+  for node in $(oc get nodes -o go-template='{{range .items}}{{$node := .}}{{range .status.conditions}}{{if eq .type "Ready"}}{{if eq .status "True"}}node/{{$node.metadata.name}}{{"\n"}}{{end}}{{end}}{{end}}{{end}}'); do
+    # shellcheck disable=SC1083
+    if ! FILE_LOCKS=$(oc debug --image="${OCDEBUGIMAGE}" "${node}" -- chroot /host sh -c "echo $SCRIPT | base64 -d > /tmp/locks.sh; chmod 755 /tmp/locks.sh; /tmp/locks.sh"); then
```
Comment (patrickeasters) [inline info/locks:15]: Should we delete this `/tmp/locks.sh` script after we're done? We've already chrooted to the host, so we'd be leaving this in the tmp directory on each node.

**Review by patrickeasters — COMMENTED** [review]
Looks good! Just a question and a nitpick.

**Review by patrickeasters — APPROVED** [review]

**Comment by dcritch** [comment]
@ptrnull did you want to take a look before merging this?

**Comment by ptrnull** [comment]
Yes, I'll check it tomorrow.

File: info/locks [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  msg "Checking for locks by pod, per node (${BLUE}using oc debug, it can take a while${NOCOLOR})"
+    fw_errors=0
+  # shellcheck disable=SC2016
+  for node in $(oc get nodes -o go-template='{{range .items}}{{$node := .}}{{range .status.conditions}}{{if eq .type "Ready"}}{{if eq .status "True"}}node/{{$node.metadata.name}}{{"\n"}}{{end}}{{end}}{{end}}{{end}}'); do
+    # shellcheck disable=SC1083
+    if ! FILE_LOCKS=$(oc debug --image="${OCDEBUGIMAGE}" "${node}" -- chroot /host sh -c "echo $SCRIPT64 | base64 -d > /tmp/locks.sh; chmod 755 /tmp/locks.sh; /tmp/locks.sh; rm -f /tmp/locks.sh"); then
+      msg "${ORANGE}Error running oc debug in ${node}${NOCOLOR}"
+    else
+      if [ -n "${FILE_LOCKS}" ]; then
+            msg "File locks found on on ${RED}${node}${NOCOLOR}"
```
Comment (ptrnull) [inline info/locks:19]: "File locks found on **on** ${RED}${node}${NOCOLOR}"

**Review by ptrnull — CHANGES_REQUESTED** [review]

**Comment by ptrnull** [comment]
It works as expected, /lgtm

```
./openshift-checks.sh -s info/locks                                                                            [13/1844]
Using kube:admin context
Checking for locks by pod, per node (using oc debug, it can take a while)
Starting pod/rna1-master-0...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-master-0...
f5ingress/f5-tmm-66456f9dbf-rw4wl                                   128
openshift-sriov-network-operator/sriov-network-config-daemon-bljxq  1
openshift-ovn-kubernetes/ovnkube-node-vv2c7                         1
Starting pod/rna1-master-1rna1 ...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-master-1...
openshift-ovn-kubernetes/ovnkube-node-sc9dg  1
Starting pod/rna1-master-2rna1 ...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-master-2...
openshift-ovn-kubernetes/ovnkube-node-hhdr9  1
Starting pod/rna1-vmaster-0-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-vmaster-0
openshift-machine-api/metal3-b7d759489-57545   24
openshift-ovn-kubernetes/ovnkube-master-cnd6v  6
openshift-ovn-kubernetes/ovnkube-node-r5kr6    1
openshift-etcd/etcd-rna1-vmaster-0             1
Starting pod/rna1-vmaster-1-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-vmaster-1
openshift-ovn-kubernetes/ovnkube-master-k4hzk  6
openshift-ovn-kubernetes/ovnkube-node-nr6jv    1
openshift-etcd/etcd-rna1-vmaster-1             1
Starting pod/rna1-vmaster-2-debug ...
To use host binaries, run `chroot /host`

Removing debug pod ...
File locks found on on node/rna1-vmaster-2
openshift-ovn-kubernetes/ovnkube-master-5d65f  6
openshift-ovn-kubernetes/ovnkube-node-wm7vd    1
openshift-etcd/etcd-rna1-vmaster-2             1
No issues found
```

**Pull request merged**